devnet-sdk/system2/faucet.go

```diff
@@ -0,0 +1,57 @@
+package system2
+
+// FaucetID identifies a Faucet by name and chainID, is type-safe, and can be value-copied and used as map key.
+type FaucetID idWithChain
+
+func (id FaucetID) String() string {
+	return idWithChain(id).string("Faucet")
+}
+
+func (id FaucetID) MarshalText() ([]byte, error) {
+	return idWithChain(id).marshalText("Faucet")
+}
+
+func (id *FaucetID) UnmarshalText(data []byte) error {
+	return (*idWithChain)(id).unmarshalText("Faucet", data)
+}
+
+func SortFaucetIDs(ids []FaucetID) []FaucetID {
+	return copyAndSort(ids, func(a, b FaucetID) bool {
+		return lessIDWithChain(idWithChain(a), idWithChain(b))
+	})
+}
+
+type Faucet interface {
+	Common
+	ID() FaucetID
+	// NewUser creates a new pre-funded user account
+	NewUser() User
```

Comment:
Tiny thing - does `Faucet` not accept some kinda balance to fund the user with? Or is it always a constant chunk of eth

Reply:
We'll probably add options to the function later, it's mostly there for scaffolding purposes, so we have something to build on in parallel later.